reana_server/reana_admin/options.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2022 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+"""Common options for the REANA administrator command line tool."""
+
+import sys
+import click
+import functools
+
+from reana_db.utils import _get_workflow_by_uuid
+
+from reana_server.utils import _get_user_by_criteria, is_uuid_v4
+
+
+def add_user_options(func):
+    """Add options to get an user by email or id."""
+
+    @click.option("-e", "--email", help="The email of the user.")
+    @click.option("--id", "id_", help="The id of the user.")
+    @functools.wraps(func)
+    def wrapper(*args, email, id_, **kwargs):
+        if id_ is not None and email is not None:
+            click.secho("Cannot provide --email and --id at the same time.", fg="red")
+            sys.exit(1)
+        user = None
+        if id_ is not None or email is not None:
+            user = _get_user_by_criteria(id_, email)
+            if not user:
+                click.secho("User not found.", fg="red")
+                sys.exit(1)
+        func(*args, user=user, **kwargs)
+
+    return wrapper
+
+
+def add_workflow_option(func):
+    """Add options to get a workflow by UUID."""
+
+    @click.option("-w", "--workflow", "workflow_uuid", help="UUID of the workflow.")
+    @functools.wraps(func)
+    def wrapper(*args, workflow_uuid, **kwargs):
+        workflow = None
+        if workflow_uuid is not None:
+            if not is_uuid_v4(workflow_uuid):
+                click.secho("Invalid workflow UUID.", fg="red")
+                sys.exit(1)
```

Comment:
Should we have an utility function for writing an error and then exiting with code `1`? We do this in many places, and it's easy to forget `sys.exit(1)`.

Reply:
I think it's a nice idea. But we should make sure it covers all the use cases, for example it could be used in `retention-rules-extend` command, but there the flag is set to be `required=True`. So it would be nice to also be able to pass additional arguments here. We could create an issue for later to extend it and to use these utilities in other cli commands